posthog/models/feature_flag.py

```diff
@@ -100,6 +100,27 @@ def get_filters(self):
                 ]
             }
 
+    @property
+    def cohort_ids(self) -> List[int]:
+        cohort_ids = []
+        for condition in self.conditions:
+            props = condition.get("properties", [])
+            for prop in props:
+                if prop.get("type", None) == "cohort":
+                    cohort_id = prop.get("value", None)
+                    if cohort_id:
+                        cohort_ids.append(cohort_id)
+        return cohort_ids
+
+    def update_cohorts(self) -> None:
+        from posthog.tasks.calculate_cohort import update_cohort
+        from posthog.tasks.cohorts_in_feature_flag import cohort_id_in_ff_key
+
+        if self.cohort_ids:
+            cache.delete(cohort_id_in_ff_key)
```

Comment:
Instead of calling `delete` here should we call https://github.com/PostHog/posthog/pull/8610/files#diff-dafed9a5d774e05093dfefdf04dd305aae5e224e2beef7720dd10a383340eb01R10 `calculate_cohort_ids_in_feature_flags` after we update the cohort in the next two lines?

Reply:
Hmm, i put the delete there as a lightweight way to trigger a recalculation of the list of ids. Because the next line, `update_cohort`, will dispatch an async task to update the cohorts. Inside the updating logic, the cache will be hit if available so if the list of cohort IDs not up to date or not empty before the update starts, postgres filling won't happen. 

For example, if we create a feature flag with a cohort filter, the cached list of cohort ids won't reflect that change unless we recalculate the list. Rather than recalculating when the feature flag is created/updated, I'm just flushing the cached value so that when the cohort calcluation happens, it'll pull fresh results when calling`get_cohort_ids_in_feature_flags`

Reply:
ah! I forgot that that was async. This is perfect then 🚢 it